jme3-code-check/src/com/jme3/gde/codecheck/hints/ReadOnlyPrimitiveHint.java

```diff
@@ -1,144 +1,91 @@
-    }
-
-    @Override
-    public List<ErrorDescription> run(CompilationInfo info, TreePath treePath) {
-
-//        MethodInvocationTree mit = (MethodInvocationTree) treePath.getLeaf();
-//        ExpressionTree select = mit.getMethodSelect();
-//        TreePath method = new TreePath(treePath, select);
-//        Element el = info.getTrees().getElement(method);
-//        TypeElement invokedClass = (TypeElement) el.getEnclosingElement();
-
-        if (info.getTrees().getElement(treePath).getAnnotation(ReadOnly.class) != null) {
+@Hint(id = "#ReadOnlyPrimitiveHint.id", displayName = "#ReadOnlyPrimitiveHint.display-name",
+        description = "#ReadOnlyPrimitiveHint.description", severity = Severity.WARNING,
+        category = "general")
+@NbBundle.Messages({
+    "ReadOnlyPrimitiveHint.display-name=This primitive is read only and should not be modified!",
+    "ReadOnlyPrimitiveHint.id=ReadOnly Primitives",
+    "ReadOnlyPrimitiveHint.description=Checks for modifications to readonly primitives. (getLocalTranslation().set())",
+    "ReadOnlyPrimitiveHint.fix-text=Remove this call"
+})
+public class ReadOnlyPrimitiveHint {
+
+    @TriggerTreeKind(Tree.Kind.METHOD_INVOCATION)
+    public static ErrorDescription hint(HintContext ctx) {
+        CompilationInfo info = ctx.getInfo();
+        TreePath treePath = ctx.getPath();
+        if (info.getTrees().getElement(treePath).getAnnotation(Annotations.ReadOnly.class) != null) {
             Tree t = treePath.getLeaf();
-            //prepare selection for removing
-            JTextComponent editor = EditorRegistry.lastFocusedComponent();
-//            Document doc = editor.getDocument();
-            SourcePositions sp = info.getTrees().getSourcePositions();
-//            int start = (int) sp.getStartPosition(info.getCompilationUnit(), t);
-            int end = (int) sp.getEndPosition(info.getCompilationUnit(), t);
+            Element el = info.getTrees().getElement(info.getTrees().getPath(info.getCompilationUnit(), t));
 
             //TODO: add more checks
-            boolean fail = false;
-            if (info.getText().length() >= end + 4) {
-                String dotText = info.getText().substring(end, end + 4);
-                if (".set".equals(dotText)) {
-                    fail = true;
-                }
-            }
-
-            if (fail) {
-                return Collections.<ErrorDescription>singletonList(
-                        ErrorDescriptionFactory.createErrorDescription(
-                        getSeverity().toEditorSeverity(),
-                        getDisplayName(),
-                        NO_FIXES,
-                        info.getFileObject(),
-                        (int) info.getTrees().getSourcePositions().getStartPosition(info.getCompilationUnit(), t),
-                        (int) info.getTrees().getSourcePositions().getEndPosition(info.getCompilationUnit(), t)));
+            if ("set".equals(el.getSimpleName().toString())) {
```

Comment:
This is wrong, it should be startsWith. Like the original tried to do, I assume. I'll fix it after I'm back from work.
```suggestion
            if (el.getSimpleName().toString().startsWith("set")) {
```